test/runtests.jl

```diff
@@ -121,32 +121,51 @@ end
                 "AWS_ACCESS_KEY_ID" => nothing
                 ) do
 
-                # Check credentials load 
+                # Check credentials load
                 config = AWSCore.aws_config()
-                creds = config[:creds]
+                creds = get_credentials(config[:creds])
 
                 @test creds.access_key_id == "TEST_ACCESS_ID"
                 @test creds.secret_key == "TEST_ACCESS_KEY"
 
+                # Check credentials refresh
+                ENV["AWS_DEFAULT_PROFILE"] = "test"
+                config = AWSCore.aws_config(
+                    creds = RenewableAWSCredentials(
+                        AWSCredentials(
+                            "EXPIRED_ACCESS_ID",
+                            "EXPIRED_ACCESS_KEY",
+                            expiry = now(UTC),
+                        ),
+                        AWSCore.dot_aws_credentials,
+                        (),
+                    )
+                )
+                creds = get_credentials(config[:creds])
+
+                @test creds.access_key_id == "TEST_ACCESS_ID"
+                @test creds.secret_key == "TEST_ACCESS_KEY"
+                @test creds.expiry > now(UTC)
+
```

Comment:
A good additional test we could do is generate some real temporary credentials which are expired. Depending on how AWS has implemented things we may be able to use expired credentials indefinitely for tests.

Reply:
I'm not quite sure how to go about that from my side since I would assume you would want to generate the expired credentials for the account we're already using for testing. Can you point me in the right direction?

Reply:
I'll note that we don't actually need "real" credentials to test the workings of `RenewableAWSCredentials` specifically since it just checks its timestamp and produces the credentials as needed. Unless the timestamp is set incorrectly, the expired credentials will not be used.

We probably should be testing `do_request` with expired credentials from an assumed role, since I'm pretty sure the retry will currently fail on master.

Reply:
I was just thinking having a real expired credential would be a good test to have. I was thinking of generating a expired credential before hand like you suggested. I'll leave it up to you to decide if that is feasible.